### ISSUE TITLE
Add support for kustomize replacements when creating custom-manifest

### ISSUE THREAD

**Issue by mathe-matician** [kbst/terraform-kubestack#325]
Right now there doesn't seem to be support for using [replacements](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/replacements/) when creating [custom-manifest](https://www.kubestack.com/framework/documentation/services/#custom-manifests).

**Comment by pst** [comment]
Thanks for the suggestion. Here's big picture what needs to be done for anyone interested to pick this up:

First needs support in the provider here: https://github.com/kbst/terraform-provider-kustomization
Then the variable can be exposed in the module here: https://github.com/kbst/catalog/tree/master/src/_terraform_module

**Comment by mathe-matician** [comment]
@pst 

> First needs support in the provider here

To clarify, isn't it already supported in the repo here? https://github.com/kbst/terraform-provider-kustomization/blob/c75a675049a3c88b285dc53600ac7d5b380d5f72/kustomize/data_source_kustomization_overlay.go#L338

So it would just need to be exposed in the `custom-manifest` module?

**Comment by pst** [comment]
You are right, replacements seem to already be supported. I guess I lost track myself.

**Comment by mathe-matician** [comment]
@pst no worries - I'm still willing to contribute to the second part you mentioned above:

> Then the variable can be exposed in the module here: https://github.com/kbst/catalog/tree/master/src/_terraform_module

**Comment by pst** [comment]
Happy to review your contribution whenever you get around to it.

If you add the variable to the` .tf` files in `src/_terraform_module` the changes will automatically be included in all future module versions built. That includes both the ones that bundle upstream like nginx or cert-manager, but also the custom-manifests module.
